esp-hal/src/dma/mod.rs

```diff
@@ -2069,6 +2097,12 @@ pub trait RegisterAccess: crate::private::Sealed {
 pub trait RxRegisterAccess: RegisterAccess {
     #[cfg(gdma)]
     fn set_mem2mem_mode(&self, value: bool);
+
+    /// Configure the channel.
+    fn configure(&self, burst_mode: bool, priority: DmaPriority) {
+        self.set_burst_mode(burst_mode);
+        self.set_priority(priority);
+    }
```

Comment:
Not sure I love this method being down here, rather than up in the `ChannelRx`/`ChannelTx`.

That aside, this should be moved up to `RegisterAccess` at least, since the RX and TX implementations are identical.